pkg/controller/maps/controller.go

```diff
@@ -0,0 +1,410 @@
+	}); err != nil {
+		return err
+	}
+
+	// Watch Pods, to ensure `status.version` and version upgrades are correctly reconciled on any change.
+	// Watching Deployments only may lead to missing some events.
+	if err := watches.WatchPods(c, NameLabelName); err != nil {
+		return err
+	}
+
+	// Watch services
+	if err := c.Watch(&source.Kind{Type: &corev1.Service{}}, &handler.EnqueueRequestForOwner{
+		IsController: true,
+		OwnerType:    &emsv1alpha1.ElasticMapsServer{},
+	}); err != nil {
+		return err
+	}
+
+	// Watch owned and soft-owned secrets
+	if err := c.Watch(&source.Kind{Type: &corev1.Secret{}}, &handler.EnqueueRequestForOwner{
+		IsController: true,
+		OwnerType:    &emsv1alpha1.ElasticMapsServer{},
+	}); err != nil {
+		return err
+	}
+	if err := watches.WatchSoftOwnedSecrets(c, emsv1alpha1.Kind); err != nil {
+		return err
+	}
+
+	// Dynamically watch referenced secrets to connect to Elasticsearch
+	return c.Watch(&source.Kind{Type: &corev1.Secret{}}, r.dynamicWatches.Secrets)
+}
+
+var _ reconcile.Reconciler = &ReconcileMapsServer{}
+
+// ReconcileMapsServer reconciles a MapsServer object
+type ReconcileMapsServer struct {
+	k8s.Client
+	operator.Parameters
+	recorder       record.EventRecorder
+	dynamicWatches watches.DynamicWatches
+	licenseChecker license.Checker
+	// iteration is the number of times this controller has run its Reconcile method
+	iteration uint64
+}
+
+func (r *ReconcileMapsServer) K8sClient() k8s.Client {
+	return r.Client
+}
+
+func (r *ReconcileMapsServer) DynamicWatches() watches.DynamicWatches {
+	return r.dynamicWatches
+}
+
+func (r *ReconcileMapsServer) Recorder() record.EventRecorder {
+	return r.recorder
+}
+
+var _ driver.Interface = &ReconcileMapsServer{}
+
+// Reconcile reads that state of the cluster for a MapsServer object and makes changes based on the state read and what is
+// in the MapsServer.Spec
+func (r *ReconcileMapsServer) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	defer common.LogReconciliationRun(log, request, "name", &r.iteration)()
+	tx, ctx := tracing.NewTransaction(ctx, r.Tracer, request.NamespacedName, "maps")
+	defer tracing.EndTransaction(tx)
+
+	// retrieve the EMS object
+	var ems emsv1alpha1.ElasticMapsServer
+	if err := association.FetchWithAssociations(ctx, r.Client, request, &ems); err != nil {
+		if apierrors.IsNotFound(err) {
+			return reconcile.Result{}, r.onDelete(types.NamespacedName{
+				Namespace: request.Namespace,
+				Name:      request.Name,
+			})
+		}
+		return reconcile.Result{}, tracing.CaptureError(ctx, err)
+	}
+
+	if common.IsUnmanaged(&ems) {
+		log.Info("Object is currently not managed by this controller. Skipping reconciliation", "namespace", ems.Namespace, "name", ems.Name)
+		return reconcile.Result{}, nil
+	}
+
+	enabled, err := r.licenseChecker.EnterpriseFeaturesEnabled()
+	if err != nil {
+		return reconcile.Result{Requeue: true, RequeueAfter: 10 * time.Second}, err
```

Comment:
I'm wondering if it's worth it to requeue the event after only 10s ? If the license is corrupted it might take some time to fix it and it might generate a lot of logs in the meantime  (and also some unnecessary load on the operator)?
```suggestion
		return reconcile.Result{}, err
```